monai/metrics/regression.py

```diff
@@ -0,0 +1,190 @@
+        reduction: {``"none"``, ``"mean"``, ``"sum"``, ``"mean_batch"``, ``"sum_batch"``,
+            ``"mean_channel"``, ``"sum_channel"``}
+            Define the mode to reduce computation result of 1 batch data. Defaults to ``"mean"``.
+
+    """
+
+    def __init__(self, reduction: Union[MetricReduction, str] = MetricReduction.MEAN) -> None:
+        super().__init__(reduction=reduction)
+        self.sq_func = partial(torch.pow, exponent=2.0)
+
+    def _compute_metric(self, y_pred: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
+        y_pred = y_pred.float()
+        y = y.float()
+
+        mse_out = compute_mean_error_metrics(y_pred, y, func=self.sq_func)
+        return mse_out
+
+
+class MAEMetric(RegressionMetric):
+    r"""Compute Mean Absolute Error between two tensors using function:
+
+    .. math::
+        \operatorname {MAE}\left(Y, \hat{Y}\right) =\frac {1}{n}\sum _{i=1}^{n}\left|y_i-\hat{y_i}\right|.
+
+    More info: https://en.wikipedia.org/wiki/Mean_absolute_error
+
+    Input `y_pred` (BCHW[D] where C is number of channels) is compared with ground truth `y` (BCHW[D]).
+    Both `y_pred` and `y` are expected to be real-valued, where `y_pred` is output from a regression model.
+
+    Args:
+        reduction: {``"none"``, ``"mean"``, ``"sum"``, ``"mean_batch"``, ``"sum_batch"``,
+            ``"mean_channel"``, ``"sum_channel"``}
+            Define the mode to reduce computation result of 1 batch data. Defaults to ``"mean"``.
+
+    """
+
+    def __init__(self, reduction: Union[MetricReduction, str] = MetricReduction.MEAN) -> None:
+        super().__init__(reduction=reduction)
+        self.abs_func = torch.abs
+
+    def _compute_metric(self, y_pred: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
+        y_pred = y_pred.float()
+        y = y.float()
+
+        mae_out = compute_mean_error_metrics(y_pred, y, func=self.abs_func)
+        return mae_out
+
+
+class RMSEMetric(RegressionMetric):
+    r"""Compute Root Mean Squared Error between two tensors using function:
+
+    .. math::
+        \operatorname {RMSE}\left(Y, \hat{Y}\right) ={ \sqrt{ \frac {1}{n}\sum _{i=1}^{n}\left(y_i-\hat{y_i}\right)^2 } } \
+        = \sqrt {\operatorname{MSE}\left(Y, \hat{Y}\right)}.
+
+    More info: https://en.wikipedia.org/wiki/Root-mean-square_deviation
+
+    Input `y_pred` (BCHW[D] where C is number of channels) is compared with ground truth `y` (BCHW[D]).
+    Both `y_pred` and `y` are expected to be real-valued, where `y_pred` is output from a regression model.
+
+    Args:
+        reduction: {``"none"``, ``"mean"``, ``"sum"``, ``"mean_batch"``, ``"sum_batch"``,
+            ``"mean_channel"``, ``"sum_channel"``}
+            Define the mode to reduce computation result of 1 batch data. Defaults to ``"mean"``.
+
+    """
+
+    def __init__(self, reduction: Union[MetricReduction, str] = MetricReduction.MEAN) -> None:
+        super().__init__(reduction=reduction)
+        self.sq_func = partial(torch.pow, exponent=2.0)
```

Comment:
not sure if this `partial` will work in a distributed evaluation case. could you follow the structure of https://github.com/Project-MONAI/MONAI/blob/45c6cf6117d2e25e88272a7001d84696e449a1cc/tests/test_handler_confusion_matrix_dist.py#L24
add a test file for RMSEMetric? the test requires two gpus on a single node.

Reply:
Sure thing, will add distributed tests for all metrics. The IterationMetric handlers for each new metric are in https://github.com/Project-MONAI/MONAI/blob/6ab5b9d40f62891bab36fb17e6c15d09fa189b17/monai/handlers/regression_metrics.py

For RMSEMetric the handler is here: https://github.com/Project-MONAI/MONAI/blob/6ab5b9d40f62891bab36fb17e6c15d09fa189b17/monai/handlers/regression_metrics.py#L87

Reply:
@wyli I have added distributed tests for all regression metrics handlers in the following file:

https://github.com/Project-MONAI/MONAI/blob/4e1db9c384026919753bca2253905cc158fe4924/tests/test_handler_regression_metrics_dist.py#L57

I tested these locally using 2 GPUs. They al pass without any changes to the metrics code. Can you please have another look and review this? From my side, it looks good to go, I would be happy to resolve any pending issues.